programs/bpf_loader/src/lib.rs

```diff
@@ -915,28 +956,33 @@ fn process_loader_upgradeable_instruction(
 
 fn common_close_account(
     authority_address: &Option<Pubkey>,
-    authority_account: &KeyedAccount,
-    close_account: &KeyedAccount,
-    recipient_account: &KeyedAccount,
+    transaction_context: &TransactionContext,
+    instruction_context: &InstructionContext,
+    first_instruction_account: usize,
     log_collector: &Option<Rc<RefCell<LogCollector>>>,
 ) -> Result<(), InstructionError> {
     if authority_address.is_none() {
         ic_logger_msg!(log_collector, "Account is immutable");
         return Err(InstructionError::Immutable);
     }
-    if *authority_address != Some(*authority_account.unsigned_key()) {
+    if *authority_address
+        != Some(*transaction_context.get_key_of_account_at_index(
+            instruction_context.get_index_in_transaction(first_instruction_account + 2)?,
```

Comment:
Here too, the magic number is getting a bit far from its source, be nice to have these defined for readability and to prevent off-by one errors over time